### ISSUE TITLE
Style/StringConcatenation shouldn't check require or require_relative calls

### ISSUE THREAD

**Issue by utkarsh2102** [rubocop/rubocop#8747]
Hello 👋🏻,

Thanks for all the great work you all put in! ❤️ 

However, whilst doing some RuboCop fixes for some projects, I found that the `Style/StringConcatenation` cop also changes the `require` and the `require_relative` calls. Should it?

I personally don't think it should or I can't think of any reasoning behind doing so. What do you think?

**Comment by utkarsh2102** [comment]
For example, the following lines are flagged as offensive:
```
spec/spec_helper.rb:5:9: C: Style/StringConcatenation: Prefer string interpolation to string concatenation.
require File.dirname(__dir__) + "/../lib/rubocop"
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
spec/spec_helper.rb:6:9: C: Style/StringConcatenation: Prefer string interpolation to string concatenation.
require File.dirname(__FILE__) + "/../lib/rubocop"
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

And are corrected to:
```
require "#{File.dirname(__dir__)}/../lib/rubocop"
require "#{File.dirname(__FILE__)}/../lib/rubocop"
```

**Comment by marcandre** [comment]
Why is this written as `require "#{__dir__}/../../lib/rubocop" and then as `require_relative '../../lib/rubocop'`?

**Comment by bbatsov** [comment]
I don't understand what's the problem with `StringConcatenation` here - the transformation seems equilavent to me.

**Comment by olliebennett** [comment]
@utkarsh2102 could you confirm whether the  `StringConcatenation` flag and autocorrect example you shared cause any problems for you? They seem equivalent and intuitive to me.

If not, I propose we close this issue 🚪

**Comment by koic** [comment]
I'll close this issue bacause it shouldn't be a problem with `Style/StringConcatenation`.
